internal/rootcoord/dml_channels.go

```diff
@@ -15,25 +15,61 @@ import (
 	"fmt"
 	"sync"
 
+	"go.uber.org/zap"
+
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/msgstream"
-	"go.uber.org/zap"
+	"github.com/milvus-io/milvus/internal/util/funcutil"
 )
 
 type dmlChannels struct {
-	core *Core
-	lock sync.RWMutex
-	dml  map[string]msgstream.MsgStream
+	core       *Core
+	lock       sync.RWMutex
+	dml        map[string]msgstream.MsgStream
+	refcnt     map[string]int64
+	namePrefix string
+	idx        int
+	idxLock    sync.Mutex
+	capacity   int
+	pool       map[string]msgstream.MsgStream
 }
 
 func newDMLChannels(c *Core) *dmlChannels {
+	const totalDmlMsgStreamNum = 64
```

Comment:
should not be a magic number here. 
We will need a configuration some where.

Reply:
add config instead